examples/cuda/hw_barrier/Makefile

```diff
@@ -0,0 +1,141 @@
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# This Makefile compiles, links, and executes examples Run `make help`
+# to see the available targets for the selected platform.
+
+################################################################################
+# environment.mk verifies the build environment and sets the following
+# makefile variables:
+#
+# LIBRAIRES_PATH: The path to the libraries directory
+# HARDWARE_PATH: The path to the hardware directory
+# EXAMPLES_PATH: The path to the examples directory
+# BASEJUMP_STL_DIR: Path to a clone of BaseJump STL
+# BSG_MANYCORE_DIR: Path to a clone of BSG Manycore
+###############################################################################
+
+REPLICANT_PATH:=$(shell git rev-parse --show-toplevel)
+
+include $(REPLICANT_PATH)/environment.mk
+
+# TEST_NAME is the basename of the executable
+TEST_NAME = main
+# KERNEL_NAME is the name of the CUDA-Lite Kernel
+KERNEL_NAME = barrier_test
+
+###############################################################################
+# Host code compilation flags and flow
+###############################################################################
+# TEST_SOURCES is a list of source files that need to be compiled
+TEST_SOURCES  = main.cpp
+
+TEST_HEADERS =
+
+DEFINES += -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -D_DEFAULT_SOURCE
+CDEFINES += 
+CXXDEFINES += 
+
+FLAGS     = -O3 -g -Wall -Wno-unused-function -Wno-unused-variable
+CFLAGS   += -std=c99 $(FLAGS)
+CXXFLAGS += -std=c++11 $(FLAGS)
+
+# compilation.mk defines rules for compilation of C/C++
+include $(EXAMPLES_PATH)/compilation.mk
+
+###############################################################################
+# Host code link flags and flow
+###############################################################################
+
+LDFLAGS =
+
+# link.mk defines rules for linking of the final execution binary.
+include $(EXAMPLES_PATH)/link.mk
+
+$(TEST_OBJECTS): $(TEST_HEADERS)
+
+###############################################################################
+# Device code compilation flow
+###############################################################################
+
+# BSG_MANYCORE_KERNELS is a list of manycore executables that should
+# be built before executing.
+
+BSG_MANYCORE_KERNELS ?= kernel.riscv
+RISCV_TARGET_OBJECTS += kernel.rvo
+RISCV_INCLUDES  +=
+RISCV_CCPPFLAGS += -D__KERNEL__
+
+TILE_GROUP_DIM_X ?= 16
+TILE_GROUP_DIM_Y ?= 8
```

Comment:
Are these still necessary with the new HW barrier?

(I'm not asking why they're not gone - just if there's some (cruft?) reason we still need them)

Reply:
Also, isn't our intent to specialized against these constants?

Reply:
Yes, I would very much like to specialize. 

But it's nice to have general code.

Reply:
Is this feedback or a wish?

Reply:
Yeah... as I think I mentioned in a manycore meeting sometime back, the amoadd barrier could just take the tile group size as arguments through registers. Nobody seemed to be on-board with that, so we went with the current approach.

Reply:
I think it's OK, how it is. Dynamically sizing a tile group would be cool. but faster code is cooler.